sunpy/map/mapbase.py

```diff
@@ -1529,20 +1529,27 @@ def draw_limb(self, axes=None, **kwargs):
         return [circ]
 
     @u.quantity_input
-    def draw_rectangle(self, bottom_left, width: u.deg, height: u.deg, axes=None, **kwargs):
+    def draw_rectangle(self, bottom_left, top_right=None, width: u.deg=None, height: u.deg=None, axes=None, **kwargs):
         """
         Draw a rectangle defined in world coordinates on the plot.
 
         Parameters
         ----------
 
-        bottom_left : `~astropy.coordinates.SkyCoord` or `~astropy.coordinates.BaseCoordinateFrame`
-            The bottom left corner of the rectangle.
+        bottom_left : `astropy.units.Quantity` or `~astropy.coordinates.SkyCoord`
+            The bottom_left coordinate of the rectangle. If a `SkyCoord` it can
+            have shape ``(2,)`` and also define ``top_right``. If specifying
+            pixel coordinates it must be given as an `~astropy.units.Quantity`
+            object with units of `~astropy.units.pixel`.
+
+        top_right : `astropy.units.Quantity` or `~astropy.coordinates.SkyCoord`
+            The top_right coordinate of the rectangle. Can only be omitted if
+            ``bottom_left`` has shape ``(2,)``.
 
-        width : `astropy.units.Quantity`
+        width : `astropy.units.Quantity` # Depreciated
             The width of the rectangle.
 
-        height : `astropy.units.Quantity`
+        height : `astropy.units.Quantity` # Depreciated
```

Comment:
```suggestion
        height : `astropy.units.Quantity` 
```
This should be in the sentence below.